operators/multiclusterobservability/controllers/placementrule/hub_info_secret.go

```diff
@@ -69,9 +69,13 @@ func generateHubInfoSecret(client client.Client, obsNamespace string,
 		}
 	}
 
-	obsApiURL := url.URL{
-		Host: obsAPIHost,
-		Path: operatorconfig.ObservatoriumAPIRemoteWritePath,
+	if !strings.HasSuffix(obsAPIHost, operatorconfig.ObservatoriumAPIRemoteWritePath) {
```

Comment:
I would add your PR description here to explain this logic. The code is already crippled with this kind of things and you don't know the reason.